File: src/main/scala/li/cil/oc/integration/gregtech/ModGregtech.java
package li.cil.oc.integration.gregtech;

import li.cil.oc.api.Driver;
import li.cil.oc.integration.IMod;
import li.cil.oc.util.mods.Mods;

public final class ModGregtech implements IMod {
    @Override
    public Mods.Mod getMod() {
        return Mods.GregTech();
    }

    @Override
    public void initialize() {
        Driver.add(new DriverEnergyContainer());
    }
}

File: src/main/scala/li/cil/oc/integration/gregtech/ModGregtech.scala
package li.cil.oc.integration.gregtech

import li.cil.oc.api
import li.cil.oc.api.Driver
import li.cil.oc.integration.ModProxy
import li.cil.oc.integration.Mods
import net.minecraftforge.common.MinecraftForge

object ModGregtech extends ModProxy {
  override def getMod = Mods.GregTech

  override def initialize() {
    api.IMC.registerToolDurabilityProvider("li.cil.oc.integration.gregtech.EventHandlerGregTech.getDurability")

    MinecraftForge.EVENT_BUS.register(EventHandlerGregTech)

    Driver.add(new DriverEnergyContainer)

    RecipeHandler.init()
  }
}